src/Models/BaseElement.php

```diff
@@ -656,15 +656,9 @@ public function getRenderTemplates($suffix = '')
      */
     public function updateFromFormData($data)
     {
-        $cmsFields = $this->getCMSFields();
-
-        foreach ($data as $field => $datum) {
-            $field = $cmsFields->dataFieldByName($field);
-
-            if (!$field) {
-                continue;
-            }
-
+        $cmsFields = $this->getCMSFields()->saveableFields();
+        foreach ($cmsFields as $fieldName => $field) {
+            $datum = $data[$fieldName] ?? null;
```

Comment:
If there's data, set to the value in the data - otherwise set to null.

This is still skipping a bunch of stuff that `$form->saveInto()` would do, but fully replicating that logic isn't in scope here. This change is the least disruptive change required to fix the bug.